File: spec/java/src/io/kaitai/struct/specwrite/TestHelloWorld.java
package io.kaitai.struct.specwrite;

import io.kaitai.struct.testwrite.HelloWorld;
import org.testng.annotations.Test;

public class TestHelloWorld extends CommonSpec {
    @Test
    public void testHelloWorld() throws Exception {
        HelloWorld r = new HelloWorld();
        r.setOne(0x50);

        assertEqualToFile(r, "fixed_struct.bin");
    }
}

File: aggregate/spec/boost_test/exception_in_pretty_xml/spec/cpp_stl_11/test_hello_world.cpp
#include <boost/test/unit_test.hpp>
#include "hello_world.h"
#include <iostream>
#include <fstream>
#include <vector>

BOOST_AUTO_TEST_CASE(test_hello_world) {
    std::ifstream ifs("src/fixed_struct.bin", std::ifstream::binary);
    kaitai::kstream ks(&ifs);
    hello_world_t r(&ks);

    BOOST_CHECK_EQUAL(r.one(), 80);
    throw std::runtime_error("special sequence ]]> in a CDATA section");
}
